### ISSUE TITLE
Logging

### ISSUE THREAD

**Issue by stevedwards** [simonbowly/math-code-guidelines#4]
Interested on yours thoughts around guidelines for logging 

I see `import logging` but not sure if you are using it at all.

**Comment by simonbowly** [comment]
Yes I probably left it behind after moving some code around (related issue for me is automatically cleaning out unused imports ... I think `isort` does this?).

I like the python docs examples for logging, so in each file:
```
import logging
logger = logging.getLogger(__name__)

...

logger.debug(...)
logger.info(...)
```

Then when you have a command-line `solver.py` file or similar, set a global logging level.  In particular this seems ideal for testing; if you have lots of logger.debug(...) lines that's a good way of tracing errors in tests.  And unlike print statements you can leave them in with (I think) no residual overhead.

Will add this to the crossdock example for comment :)
